### PR TITLE
[Dist] fix is_node check according to policy

### PR DESCRIPTION
**Pull request by Rhett-Ying**
## Checklist
Please feel free to remove inapplicable items for your PR.
- [ ] The PR title starts with [$CATEGORY] (such as [NN], [Model], [Doc], [Feature]])
- [ ] I've leverage the [tools](https://docs.google.com/document/d/1iHyj7zlmygKSk5gBPsqIqL5ASPzJSPREaNT_QdsiYA4/edit) to beautify the python and c++ code.
- [ ] The PR is complete and small, read the [Google eng practice (CL equals to PR)](https://google.github.io/eng-practices/review/developer/small-cls.html) to understand more about small PR. In DGL, we consider PRs with less than 200 lines of core code change are small (example, test and documentation could be exempted).
- [ ] All changes have test coverage
- [ ] Code is well-documented
- [ ] To the best of my knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change
- [ ] Related issue is referred in this PR
- [ ] If the PR is for a new model/paper, I've updated the example index [here](../examples/README.md).

### CODE REVIEW

File: python/dgl/distributed/graph_partition_book.py [hunk 1]
```diff
@@ -1293,7 +1293,7 @@ def partition_book(self):
 
     def get_data_name(self, name):
         """Get HeteroDataName"""
-        is_node = NODE_PART_POLICY in self.policy_str
+        is_node = self.policy_str.startswith(NODE_PART_POLICY)
```
Comment (jermainewang) [inline python/dgl/distributed/graph_partition_book.py:1296]: Please also fix `EDGE_PART_POLICY`?

Comment (jermainewang) [reply python/dgl/distributed/graph_partition_book.py:1296]: In general, we should replace all occurrances of `XXX_PART_POLICY` with `startswith` or create a `is_node_policy` util for that.

**Review by jermainewang — CHANGES_REQUESTED** [review]
See my comments.

File: python/dgl/distributed/graph_partition_book.py (hunk 1 above)
Comment (Rhett-Ying) [reply python/dgl/distributed/graph_partition_book.py:1296]: agree

Comment (Rhett-Ying) [reply python/dgl/distributed/graph_partition_book.py:1296]: How about enforcing the policy has to be one of `NODE_PART_POLICY` and `EDGE_PART_POLICY`. DGL supports these 2 only for now.

Comment (jermainewang) [reply python/dgl/distributed/graph_partition_book.py:1296]: Sure. Having more sanity check is always welcomed.

Comment (Rhett-Ying) [reply python/dgl/distributed/graph_partition_book.py]: added.

**Review by jermainewang — APPROVED** [review]

**Pull request merged**